src/gsMSplines/gsWeightMapper.cpp

```diff
@@ -0,0 +1,210 @@
+/** @file gsWeightMapper.cpp
```

Comment:
The class should be templated instead of using real_t hardcoded.
This also implies to rename this file as gsWeighMapper.hpp
and also add a new instantiation file gsWeighMapper_.cpp